src/stores/installed-apps-api.ts

```diff
@@ -0,0 +1,39 @@
+import { createApi, fetchBaseQuery } from '@reduxjs/toolkit/query/react';
+import { getIpcApi } from 'src/lib/get-ipc-api';
+
+export type InstalledAppsState = InstalledApps & InstalledTerminals;
+
+export const installedAppsInitialState: InstalledAppsState = {
+	vscode: false,
+	phpstorm: false,
+	webstorm: false,
+	windsurf: false,
+	cursor: false,
+	terminal: false,
+	iterm: false,
+	ghostty: false,
+	warp: false,
+};
+
+export const installedAppsApi = createApi( {
+	reducerPath: 'installedAppsApi',
+	baseQuery: fetchBaseQuery(),
+	tagTypes: [ 'InstalledApps' ],
+	endpoints: ( builder ) => ( {
+		getInstalledApps: builder.query< InstalledAppsState, void >( {
+			queryFn: async () => {
+				try {
+					const installedApps = await getIpcApi().getInstalledApps();
+					return { data: installedApps as InstalledAppsState };
+				} catch ( error ) {
+					return {
+						data: installedAppsInitialState,
+					};
+				}
+			},
+			providesTags: [ 'InstalledApps' ],
+		} ),
+	} ),
+} );
+
+export const { useGetInstalledAppsQuery } = installedAppsApi;
```

Comment:
```suggestion
export const { useGetInstalledAppsQuery } = installedAppsApi;

export const selectInstalledEditors = createSelector(
	[ ( data?: InstalledAppsState ) => data ],
	( installedApps ) => {
		const entries = Object.entries( supportedEditorConfig ) as [
			SupportedEditor,
			SupportedEditorConfig,
		][];

		return entries.filter( ( [ editor ] ) => installedApps && installedApps[ editor ] );
	}
);

export const selectUninstalledEditors = createSelector(
	[ ( data?: InstalledAppsState ) => data ],
	( installedApps ) => {
		const entries = Object.entries( supportedEditorConfig ) as [
			SupportedEditor,
			SupportedEditorConfig,
		][];

		return entries.filter( ( [ editor ] ) => ! installedApps || ! installedApps[ editor ] );
	}
);

```
Consider this a suggestion, but I would add selectors here to select the installed editors. The `createSelector` API isn't the most intuitive, but it gives us built-in caching that's useful.